src/gretel_client/client.py

```diff
@@ -259,6 +295,7 @@ def _write_records(
             t = threading.Thread(
                 target=self._write_record_batch,
                 args=(project, write_queue, error_list, thread_event),
+                kwargs=kwargs,
```

Comment:
does this need to be `**kwargs`?

Reply:
I believe so. `Thread.args` and `Thread.kwargs` represents `*args` and `**kwargs` when constructing the call to `target`. I was losing the named arguments when passing `**kwards` into `args`.